websockets-protocol-binding.md

```diff
@@ -0,0 +1,164 @@
+
+This specification does not prescribe rules constraining the use or handling of
+specific [HTTP target resource][rfc7230-section-5-1] to establish the WebSocket
+upgrade.
+
+This specification prescribe rules constraining the [WebSockets
+Subprotocols][rfc6455-section-5-1] in order to reach agreement on the event
+format to use, in order to exchange serialized CloudEvents.
+
+Events are sent as WebSocket messages, serialized using an [event
+format][ce-event-format].
+
+### 1.3. Content Modes
+
+The [CloudEvents specification][ce-message] defines two content modes for
+transferring events: _structured_ and _binary_.
+
+Because of the nature of WebSockets messages, this specification supports only
+_structured_ data mode, hence event metadata attributes and event data are
+placed into the WebSockets messages using an [event format][ce-event-format].
+
+The [event format][ce-event-format] to be used in a full-duplex WebSocket stream
+is agreed during the [handshake](#14-handshake) and cannot change during the
+same stream.
+
+### 1.4. Handshake
+
+The [opening handshake][rfc6455-section-1-3] MUST follow the set of rules
+specified in the [RFC6455][rfc6455-section-4].
+
+In addition, the client MUST include, in the opening handshake, the
+[`Sec-WebSocket-Protocol` header][rfc6455-section-1-9]. The client MUST include
+in this header one or more
+[CloudEvents subprotocols](#15-cloudevents-subprotocols), depending on the
+subprotocols the client supports.
+
+The server MUST reply with the chosen CloudEvents subprotocol using the
+`Sec-WebSocket-Protocol` header. If the server doesn't support any of the
+subprotocols included in the opening handshake, the server response should not
+contain any `Sec-WebSocket-Protocol` header.
+
+#### 1.4.1 Example
+
+Example client request to begin the opening handshake:
+
+```text
+GET /events HTTP/1.1
+Host: server.example.com
+Upgrade: websocket
+Connection: Upgrade
+Sec-WebSocket-Key: x3JJHMbDL1EzLkh9GBhXDw==
+Sec-WebSocket-Protocol: cloudevents.json, cloudevents.avro
+Sec-WebSocket-Version: 13
+Origin: http://example.com
+```
+
+Example server response:
+
+```text
+HTTP/1.1 101 Switching Protocols
+Upgrade: websocket
+Connection: Upgrade
+Sec-WebSocket-Accept: HSmrc0sMlYUkAGmm5OPpG2HaGWk=
+Sec-WebSocket-Protocol: cloudevents.json
+```
+
+### 1.5. CloudEvents Subprotocols
+
+This specification maps a WebSocket subprotocol to each defined event format in
+the CloudEvents specification, following the guidelines discussed in the
+[RFC6455][rfc6455-section-1-9]. For each subprotocol, senders MUST use the
+specified WebSocket frame type:
+
+| Subprotocol        | Event format                     | Frame Type |
+| ------------------ | -------------------------------- | ---------- |
+| `cloudevents.json` | [JSON event format][json-format] | Text       |
+| `cloudevents.avro` | [AVRO event format][avro-format] | Binary     |
+
+All implementations of this specification MUST support the [JSON event
```

Comment:
With all implementations, did you mean client and server side?
Could you please elaborate why JSON event format MUST be supported?

Honestly I would delete this part and leave it to those implementing the protocol.

Reply:
Clemens clarified in todays call, that JSON is mandated anyhow - I wouldn't mind when you @slinkydeveloper just close that comment.